src/common/fetchClient.ts

```diff
@@ -90,17 +90,32 @@ export default class FetchClient {
   private buildUrl(path: string, params?: FetchRequestConfig["params"]): URL {
     const url = new URL(this.config.baseURL + path);
 
-    if (params) {
-      Object.entries(params).forEach(([key, value]) => {
+    function appendParams(key: string, value: any, parentKey?: string) {
+      const fullKey = parentKey ? `${parentKey}[${key}]` : key;
+
+      if (
+        typeof value === "object" &&
+        value !== null &&
+        !(value instanceof Date) &&
+        !(value instanceof Array)
+      ) {
```

Comment:
I'm sure this is needed just curious what this thorough of a check is doing?

Reply:
oh this is just making sure this is only processing plain objects (`{}`) - is there a better way to do that?